src/coreclr/src/tools/Common/TypeSystem/Common/MetadataVirtualMethodAlgorithm.cs

```diff
@@ -99,6 +102,42 @@ public Enumerator GetEnumerator()
                 return new Enumerator(_members);
             }
 
+            public IEnumerable<MethodDesc> MethodsRequiringSlotUnification
+            {
+                get
+                {
+                    foreach (var method in _methodsRequiringSlotUnification)
+                    {
+                        if (method == null)
```

Comment:
How could `null` appear in the _methodsRequiringSlotUnification?

Reply:
It's `Array.Resize`d a couple lines down - null entries mark unused space left from that.

Maybe this could use an `ArrayBuilder<MethodDesc>` to wrap the detail that the array is only usable up to `_methodsRequiringSlotUnificationCount`.